tests/test-chat-template-chaton.cpp

```diff
@@ -0,0 +1,179 @@
+#include <iostream>
+#include <string>
+#include <vector>
+#include <sstream>
+
+#undef NDEBUG
+#include <cassert>
+
+#include "llama.h"
+#include "chaton.hpp"
+
+
+std::vector<std::string> templateIds = { "llama2", "llama3", "chatml",
```

Comment:
```suggestion
std::vector<std::string> g_template_ids = { "llama2", "llama3", "chatml",
```

Reply:
[habit vestige] I normally use snake case for function names and camel case for variables/args etal, to help differentiate between them from name itself in general, for simplicity across different languages independent of any module/oops/indirect data+logic bonhomie etal, but I assume you prefer the snake case convention. Is it needed only for global variables, or does llama.cpp enforce it even for local variables within functions/class etal.